ComSemApp/templates/ComSemApp/student/assessment.html

```diff
@@ -80,6 +97,80 @@ <h2 id="timer-text" class="show-expression"></h2>
   });
 }
 
+function timer_promise(sentence){
+  let reading_check = timer(sentence.length/2, READING_OFFSET);
+  //Once reading timer done this starts
+  reading_check.then(() => {
+    $("#timer-bar").removeClass("bg-info").addClass("bg-danger")
+    //Runs the recording timer
+
+    //starts recording
+    recorder.start();
+    let recording_check = timer(sentence.length/2, RECORDING_OFFSET);
+    //Once recording timer done this starts
+    recording_check.then(() => { 
+      // stops recording
+      recorder.stop();
+      // keeps next problem button disabled if no more problems left
+      if (curr_problem_index < problems.length -1 ){
+        toggle_enabled_next();
+      }
+      
+
+      // updates the progress bar
+      process_response();
+      save_in_form();
+      $("#expression-text").html('Problem Complete!');
+    });
+  });
+}
+
+function next_problem(){
+  toggle_enabled_next();
+  curr_problem_index = curr_problem_index + 1;
+  var sentence = problems[curr_problem_index]['formulation'];
+
+  $("#expression-text").html(sentence);
+  console.log("changed expression to " + sentence);
+  $("#timer-bar").addClass("bg-info").removeClass("bg-danger")
+  timer_promise(sentence);
+}
+
+function toggle_enabled_next(){
+  if ($("#next").hasClass("disabled")){
+    $("#next").removeClass("disabled");
+  }
+  else{
+    $("#next").addClass("disabled");
+  }
+}
+
+function save_in_form(){
+  let attemptFormData = new FormData();
+  attemptFormData.append( 'expression', problems.id);
+  attemptFormData.append('student', student);
+  attemptFormData.append('audio', audioReformulationBlob);
+  var csrf_token = '{{ csrf_token }}';
+  $.ajax({
+        type: "POST",
+        url: $("#save_attempt_url").val(),
+        data: attemptFormData,
+        processData: false,
+        contentType: false,
+        headers: {
+           'X-CSRFToken': csrf_token
+        },
+        success: function(response){
+            cs_notification('success', "Expression Saved")
```

Comment:
Just noticed this while doing some usability testing: this says "Expression Saved" when it should say something more along the lines of "Attempt Saved"